src/cbortojson.c

```diff
@@ -329,9 +329,10 @@ static CborError escape_text_string(char **str, size_t *alloc, size_t *offsetp,
         return CborErrorDataTooLarge;
     }
     if (!alloc || needed > *alloc) {
-        buf = cbor_realloc(buf, needed);
-        if (!buf)
+        char *tmp = cbor_realloc(buf, needed);
+        if (!tmp)
             return CborErrorOutOfMemory;
+        buf = tmp;
```

Comment:
Even if there is a problem with freeing, this patch isn't solving it. `buf` is a local variable in this function, so adding yet another temporary is not going to help. This code will compile to exactly the same assembly before and after the change.

Reply:
@thiagomacieira, how do you propose a change so that its more secure?